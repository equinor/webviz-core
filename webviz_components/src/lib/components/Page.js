import React from 'react';
import PropTypes from 'prop-types';

const Page = ({id, children, title}) => (
    <div style={{margin: '20px'}} id={id} key={title}>
        {children}
    </div>
);

Page.defaultProps = {
    children: null,
};

Page.propTypes = {
    /**
     * The ID of this component, used to identify dash components
     * in callbacks. The ID needs to be unique across all of the
     * components in an app.
     */
    id: PropTypes.string.isRequired,
    /**
     * The title of this page. Will be displayed in the menu.
     */
    title: PropTypes.string.isRequired,
    /**
     * The content of the page.
     */
    children: PropTypes.node,
};

export default Page;
