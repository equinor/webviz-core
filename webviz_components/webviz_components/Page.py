# AUTO GENERATED FILE - DO NOT EDIT

from dash.development.base_component import Component, _explicitize_args


class Page(Component):
    """A Page component.


Keyword arguments:
- children (a list of or a singular dash component, string or number; optional): The content of the page.
- id (string; required): The ID of this component, used to identify dash components
in callbacks. The ID needs to be unique across all of the
components in an app.
- title (string; required): The title of this page. Will be displayed in the menu."""
    @_explicitize_args
    def __init__(self, children=None, id=Component.REQUIRED, title=Component.REQUIRED, **kwargs):
        self._prop_names = ['children', 'id', 'title']
        self._type = 'Page'
        self._namespace = 'webviz_components'
        self._valid_wildcard_attributes =            []
        self.available_properties = ['children', 'id', 'title']
        self.available_wildcard_properties =            []

        _explicit_args = kwargs.pop('_explicit_args')
        _locals = locals()
        _locals.update(kwargs)  # For wildcard attrs
        args = {k: _locals[k] for k in _explicit_args if k != 'children'}

        for k in ['id', 'title']:
            if k not in args:
                raise TypeError(
                    'Required argument `' + k + '` was not specified.')
        super(Page, self).__init__(children=children, **args)
