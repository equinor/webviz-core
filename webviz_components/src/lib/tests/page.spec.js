import React from 'react';
import renderer from 'react-test-renderer';
import Page from '../components/Page';

describe('Page', () => {
    it('should render Page', () => {
        const tree = renderer.create(
            <Page id="page" title="Page">
                <h1>Header</h1>
                <p>Some text here</p>
                <p>Some more text here</p>
            </Page>
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});
