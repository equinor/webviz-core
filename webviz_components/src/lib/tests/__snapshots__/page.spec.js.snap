// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render Page 1`] = `
<div
  id="page"
  style={
    Object {
      "margin": "20px",
    }
  }
>
  <h1>
    Header
  </h1>
  <p>
    Some text here
  </p>
  <p>
    Some more text here
  </p>
</div>
`;
